tables/table.py

```diff
@@ -1364,8 +1402,12 @@ def will_query_use_indexing(self, condition, condvars=None):
         idxcols = [condvars[var].pathname for var in compiled.index_variables]
         return frozenset(idxcols)
 
-    def where(self, condition, condvars=None,
-              start=None, stop=None, step=None):
+    def where(self,
+              condition: str,
+              condvars: Optional[dict[str, Union["Column", np.ndarray]]]=None,
+              start: Optional[str]=None,
+              stop: Optional[str]=None,
+              step: Optional[str]=None) -> Iterator:
```

Comment:
See the comment on `__iter__`

Reply:
As recommended changed to `Iterator[tableextension.Row]`.